contracts/utils/SignatureValidator.sol

```diff
@@ -99,4 +100,38 @@ contract SignatureValidator {
 
     return signer;
   }
+
+ /**
+   * @notice Returns true if the provided signature is valid for the given signer.
+   * @dev Supports SignatureType.EIP712, SignatureType.EthSign, and ERC1271 signatures
+   * @param _hash      Hash that was signed
+   * @param _signer    Address of the signer candidate
+   * @param _signature Signature byte array
+   */
+  function isValidSignature(
+    bytes32 _hash,
+    address _signer,
+    bytes memory _signature
+  ) internal view returns (bool valid) {
+    uint256 signatureType = uint8(_signature[_signature.length - 1]);
+
+    if (signatureType == SIG_TYPE_EIP712 || signatureType == SIG_TYPE_ETH_SIGN) {
```

Comment:
What is the point of processing these types here? It looks the same as the `if (flag == FLAG_SIGNATURE)` part. And the signature size should also be the same.

Reply:
It's for completion, most of the implementation work is on `recoverSigner` and it makes handling of signatures a little bit more flexible. It may not be worth it tho.